### ISSUE TITLE
Beamer table of contents

### ISSUE THREAD

**Issue by prowlett** [coursebuilder-ncl/makecourse#48]
In Beamer, I can provide a slide giving a table of contents like this. 

```
\begin{frame}
\frametitle{Contents}
\tableofcontents
\end{frame}
```

In the HTML output of `makecourse`, the `\tableofcontents` command is ignored and I just get `<h2>Contents</h2><h2>Next section or frame title</h2>`, which looks strange. I guess there is no need to render the table of contents, since the HTML page has the table of contents navigation built in, but it would be nice if a slide whose main content (i.e. apart from the title) was only `\tableofcontents` would be omitted from the HTML display, in favour of the built-in table of contents.

**Comment by prowlett** [comment]
I should say I consider this very low priority - I can work around it, or users can just put up with the extra heading.

**Comment by prowlett** [comment]
For example, here is a workaround I knocked up for the moment.

```
\documentclass{beamer}

\usepackage{makecourse}
\ifplastex
\newcommand{\putcontentshere}{}
\else
\newcommand{\putcontentshere}{\begin{frame}
\frametitle{Contents}
\tableofcontents
\end{frame}}
\fi

\begin{document}

\putcontentshere

% Slides here 

\end{document}
```

**Comment by georgestagg** [comment]
The `\tableofcontents` command has been implemented for beamer slides in commit a0164e85e586f4d134e2d5cb4ebfba828aa044f2

This could come in handy if the sidebar is disabled, so for the moment the output is included in the HTML version in full whenever the beamer documentclass is used.

I think your idea to hide the frame in the HTML using `\ifplastex` is the right way to go when you do want to suppress the output. I would recommend using something like,
```
\ifplastex
\else
\begin{frame}
\frametitle{Contents}
\tableofcontents
\end{frame}
\fi
```

**Comment by prowlett** [comment]
Works well, thanks! Yes, your `\ifplastex` is better!
